### PR TITLE
Fine graining of graphql api

### PR DESCRIPTION
**Pull request by Rembane**

### CODE REVIEW

File: src/Model/KarenGraphQLApi.hs [hunk 1]
```diff
@@ -176,6 +168,26 @@ fetchMenu lang restaurantUUID day = do
       ]
     ]
 
+-- | Fetches menus from Kåren's GraphQL API.
+-- Parameters: Language, RestaurantUUID, day
+fetchMenu
+  :: (Applicative m, MonadIO m, MonadReader ClientContext m, MonadThrow m)
```
Comment (Jassob) [inline src/Model/KarenGraphQLApi.hs]: We don't need the Applicative constraint on `m` right?

**Review by Jassob — CHANGES_REQUESTED** [review]
Remove unnecessary type constraint, then it's ready to merge! :+1:

File: src/Model/KarenGraphQLApi.hs (hunk 1 above)
Comment (Rembane) [reply src/Model/KarenGraphQLApi.hs]: We did, but for some interesting reason we don't anymore. Changes are pushed.

**Review by Jassob — APPROVED** [review]

**Comment by Jassob** [comment]
When fixing review comments I'm used to simply amending/"fixuping" the responsible commit rather than creating a "fix comments" commit and I think that is a good workflow that we should adopt here. At the very least in case we have build errors or failing tests.

**Comment by Rembane** [comment]
I don't follow. Should I in the future remove comments from intermediate commits in the final squashed commit message?

**Comment by Jassob** [comment]
I mean that I would have preferred the last commit to have been squashed with the first commit. 

The goal is to keep the history as short and relevant as possible. Whether we removed a type constraint in the commit that refactored its function or in a later commit does not really matter since both commits are rather small, semantically related changes, but the history does not become clearer by removing it in later commit (unless it was a more controversial change, in that case a new commit together with a clear and descriptive commit message is absolutely the right way) so I think it should have been done in one commit, namely the first. :smile:

**Comment by Jassob** [comment]
Aha, well, my comment was more related to when merging of PRs are made by rebasing them onto master.
When squashing it does not matter, sorry :)

**Comment by Rembane** [comment]
No worries. :D

**Pull request merged**